evalml/tests/pipeline_tests/test_pipelines.py

```diff
@@ -1991,3 +1996,50 @@ def test_score_error_when_custom_objective_not_instantiated(problem_type, logist
     # Verify no exception when objective properly specified
     if is_binary(problem_type):
         pipeline.score(X, y, objectives=[CostBenefitMatrix(1, 1, -1, -1), "F1"])
+
+
+def test_binary_pipeline_string_target_thresholding(make_data_type, X_y_binary):
+    class BinaryClass(BinaryClassificationPipeline):
+        component_graph = ['Imputer', 'One Hot Encoder', 'Logistic Regression Classifier']
+
+    X, y = X_y_binary
+    X = make_data_type('ww', (X))
+    y = make_data_type('ww', pd.Series([f"String value {i}" for i in y]))
+    objective = get_objective("F1", return_instance=True)
+    pipeline = BinaryClass(parameters={"Logistic Regression Classifier": {"n_jobs": 1}})
+    pipeline.fit(X, y)
+    assert pipeline.threshold is None
+    pred_proba = pipeline.predict_proba(X, y).iloc[:, 1]
+    pipeline.optimize_threshold(X, y, pred_proba, objective)
+    assert pipeline.threshold is not None
+
+
+def test_pipeline_thresholding_errors(make_data_type, X_y_multi, X_y_binary):
```

Comment:
Major nitpick but maybe we can mock fit/predict here since they're not necessary? :)